File: src/main/dart/net/coderextreme/data/DesignPatternsApparelMedicalSkinLayers.dart
// Copyright 2021 by John Carlson.  All rights reserved.
import 'dart:io';
import '../x3d.dart';
var X3D0 =  X3D(

      profile_ : SFString('Immersive'),
      version_ : SFString('4.0'),
      head_ : 
        head(
          meta_ : [
            meta(
              name_ : SFString('title'),
              content_ : SFString('DesignPatternsApparelMedicalSkinLayers.x3d')),

            meta(
              name_ : SFString('description'),
              content_ : SFString('Design patterns for skin and apparel using HAnim2 standard in X3D4')),

            meta(
              name_ : SFString('creator'),
              content_ : SFString('Don Brutzman')),

            meta(
              name_ : SFString('creator'),
              content_ : SFString('Joe D. Williams')),

            meta(
              name_ : SFString('creator'),
              content_ : SFString('Dick Puk')),

            meta(
              name_ : SFString('created'),
              content_ : SFString('23 December 2022')),

            meta(
              name_ : SFString('modified'),
              content_ : SFString('2 July 2023')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('DesignPatternsApparelVariations.txt')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('HAnim2 Part 1, HAnim architecture, 4.3 Humanoid object https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/concepts.html#HumanoidObject')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('HAnim2 Part 1, HAnim architecture, E.4 Multiple humanoids per file https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/Guidelines.html#MultipleHumanoidsPerFile')),

            meta(
              name_ : SFString('reference'),
              content_ : SFString('X3D4 Architecture, clause 26 HAnim component, 26.3.2 HAnimHumanoid https://www.web3d.org/specifications/X3Dv4Draft/ISO-IEC19775-1v4-DIS/Part01/components/hanim.html#HAnimHumanoid')),

            meta(
              name_ : SFString('warning'),
              content_ : SFString('Under development. This template example does not produce renderable HAnim models.')),

            meta(
              name_ : SFString('identifier'),
              content_ : SFString('https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Skin/DesignPatternsApparelMedicalSkinLayers.x3d')),

            meta(
              name_ : SFString('generator'),
              content_ : SFString('X3D-Edit 4.0, https://savage.nps.edu/X3D-Edit'))]),
      Scene_ : 
        Scene(
          children_ : [
            WorldInfo(
              title_ : SFString('HAnimHumanoid skin design patterns for apparel, medical')),

            Background(
              skyColor_ : MFColor([SFColor(0), SFColor(0.6), SFColor(0.6)])),

            Group(
              DEF_ : SFString('MultipleHumanoids'),
              metadata_ : [
                MetadataString(
                  name_ : SFString('HAnimArchitecture'),
                  reference_ : SFString('https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/Guidelines.html#MultipleHumanoidsPerFile'),
                  value_ : MFString([SFString("E.4 Multiple humanoids per file")]))]),

            Viewpoint(
              DEF_ : SFString('ViewHelpText'),
              description_ : SFString('Select text to see website'),
              position_ : SFVec3f([SFDouble(0), SFDouble(0), SFDouble(12)])),

            Anchor(
              description_ : SFString('DesignPatternsApparelMedicalSkinLayers.x3d explores potential apparel approaches'),
              url_ : MFString([SFString("https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Skin/DesignPatternsApparelMedicalSkinLayersIndex.html")]),
              parameter_ : MFString([SFString("target=blank")]),
              children_ : [
                Shape(
                  appearance_ : 
                    Appearance(
                      material_ : 
                        Material(
                          diffuseColor_ : SFColor([SFDouble(0.9), SFDouble(0.9), SFDouble(0.9)]))),
                  geometry_ : 
                    Text(
                      string_ : MFString([SFString("DesignPatternsApparelMedicalSkinLayers.x3d"), SFString(""), SFString("explores potential apparel approaches")]),
                      fontStyle_ : 
                        FontStyle(
                          family_ : MFString([SFString("SANS")]),
                          style_ : SFString('BOLD'),
                          size_ : 0.6,
                          justify_ : MFString([SFString("MIDDLE"), SFString("MIDDLE")])))),

                Shape(
                  appearance_ : 
                    Appearance(
                      material_ : 
                        Material(
                          transparency_ : 1)),
                  geometry_ : 
                    Box(
                      size_ : SFVec3f([SFDouble(11), SFDouble(2), SFDouble(0.001)])))])]));
void main() { exit(0); }

File: src/main/nashorn/net/coderextreme/data/DesignPatternsApparelMedicalSkinLayers.js
load('X3Dautoclass.js');
var ConfigurationProperties = Packages.org.web3d.x3d.jsail.ConfigurationProperties;
ConfigurationProperties.showDefaultAttributes = false;
ConfigurationProperties.xsltEngine = ConfigurationProperties.XSLT_ENGINE_NATIVE_JAVA;
ConfigurationProperties.deleteIntermediateFiles = false;
ConfigurationProperties.setStripTrailingZeroes(true);
      var X3D0 =  new X3D().setProfile("Immersive").setVersion("4.0")
      .setHead(new head()
        .addMeta(new meta().setName("title").setContent("DesignPatternsApparelMedicalSkinLayers.x3d"))
        .addMeta(new meta().setName("description").setContent("Design patterns for skin and apparel using HAnim2 standard in X3D4"))
        .addMeta(new meta().setName("creator").setContent("Don Brutzman"))
        .addMeta(new meta().setName("creator").setContent("Joe D. Williams"))
        .addMeta(new meta().setName("creator").setContent("Dick Puk"))
        .addMeta(new meta().setName("created").setContent("23 December 2022"))
        .addMeta(new meta().setName("modified").setContent("2 July 2023"))
        .addMeta(new meta().setName("reference").setContent("DesignPatternsApparelVariations.txt"))
        .addMeta(new meta().setName("reference").setContent("HAnim2 Part 1, HAnim architecture, 4.3 Humanoid object https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/concepts.html#HumanoidObject"))
        .addMeta(new meta().setName("reference").setContent("HAnim2 Part 1, HAnim architecture, E.4 Multiple humanoids per file https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/Guidelines.html#MultipleHumanoidsPerFile"))
        .addMeta(new meta().setName("reference").setContent("X3D4 Architecture, clause 26 HAnim component, 26.3.2 HAnimHumanoid https://www.web3d.org/specifications/X3Dv4Draft/ISO-IEC19775-1v4-DIS/Part01/components/hanim.html#HAnimHumanoid"))
        .addMeta(new meta().setName("warning").setContent("Under development. This template example does not produce renderable HAnim models."))
        .addMeta(new meta().setName("identifier").setContent("https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Skin/DesignPatternsApparelMedicalSkinLayers.x3d"))
        .addMeta(new meta().setName("generator").setContent("X3D-Edit 4.0, https://savage.nps.edu/X3D-Edit")))
      .setScene(new Scene()
        .addChild(new WorldInfo().setTitle("HAnimHumanoid skin design patterns for apparel, medical"))
        .addChild(new Background().setSkyColor(Java.to([0,0.6,0.6], Java.type("double[]"))))
        .addChild(new Group().setDEF("MultipleHumanoids")
          .setMetadata(new MetadataString().setName("HAnimArchitecture").setReference("https://www.web3d.org/documents/specifications/19774/V2.0/Architecture/Guidelines.html#MultipleHumanoidsPerFile").setValue(Java.to(["E.4 Multiple humanoids per file"], Java.type("java.lang.String[]")))))
        .addChild(new Viewpoint().setDEF("ViewHelpText").setDescription("Select text to see website").setPosition(Java.to([0,0,12], Java.type("double[]"))))
        .addChild(new Anchor().setDescription("DesignPatternsApparelMedicalSkinLayers.x3d explores potential apparel approaches").setUrl(Java.to(["https://www.web3d.org/x3d/content/examples/HumanoidAnimation/Skin/DesignPatternsApparelMedicalSkinLayersIndex.html"], Java.type("java.lang.String[]"))).setParameter(Java.to(["target=blank"], Java.type("java.lang.String[]")))
          .addChild(new Shape()
            .setAppearance(new Appearance()
              .setMaterial(new Material().setDiffuseColor(Java.to([0.9,0.9,0.9], Java.type("double[]")))))
            .setGeometry(new Text().setString(Java.to(["DesignPatternsApparelMedicalSkinLayers.x3d","","explores potential apparel approaches"], Java.type("java.lang.String[]")))
              .setFontStyle(new FontStyle().setFamily(Java.to(["SANS"], Java.type("java.lang.String[]"))).setStyle("BOLD").setSize(0.6).setJustify(Java.to(["MIDDLE","MIDDLE"], Java.type("java.lang.String[]"))))))
          .addChild(new Shape()
            .setAppearance(new Appearance()
              .setMaterial(new Material().setTransparency(1)))
            .setGeometry(new Box().setSize(Java.to([11,2,0.001], Java.type("double[]")))))))      ;
    X3D0.toFileJSON("../data/DesignPatternsApparelMedicalSkinLayers.new.graal.json");
